onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc

```diff
@@ -0,0 +1,106 @@
+  // Add operator related.
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related.
+ private:
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                         const logging::Logger& logger) const override;
+};
+
+Status NormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                                     const Node& node,
+                                                     const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  emscripten::val input = model_builder.GetOperand(input_defs[0]->Name());
+  emscripten::val options = emscripten::val::object();
+  options.set("scale", model_builder.GetOperand(input_defs[1]->Name()));
+
+  if (input_defs.size() == 3) {
+    // Inputs contain optional bias
+    options.set("bias", model_builder.GetOperand(input_defs[2]->Name()));
+  }
+
+  options.set("epsilon", helper.Get("epsilon", 1e-05f));
+
+  std::vector<int64_t> input_shape;
+  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get input shape");
+  const auto rank = input_shape.size();
+
+  NodeAttrHelper helper(node);
+  int64_t axis = helper.Get("axis", -1);
+  axis = HandleNegativeAxis(axis, rank);
+  std::vector<int32_t> axes{static_cast<int32_t>(axis)};
+  options.set("axes", emscripten::val::array(axes));
+
+  emscripten::val output = model_builder.GetBuilder().call<emscripten::val>("meanVarianceNormalization", input, options);
+  model_builder.AddOperand(node.OutputDefs()[0]->Name(), std::move(output));
+
+  return Status::OK();
+}
+
+// Operator support related.
+
+bool NormalizationOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                                               const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  std::vector<int64_t> input_shape;
+  if (!GetShape(*input_defs[0], input_shape, logger)) {
+    LOGS(logger, VERBOSE) << "Cannot get input shape.";
+    return false;
+  }
+  const auto rank = input_shape.size();
+
+  NodeAttrHelper helper(node);
+  int64_t axis = helper.Get("axis", -1);
+  axis = HandleNegativeAxis(axis, rank);
+
+  const auto& scale_name = input_defs[1]->Name();
+  if (!Contains(initializers, scale_name)) {
+    LOGS(logger, VERBOSE) << "The scale must be a constant initializer.";
+    return false;
+  }
+
+  if (input_defs.size() == 3) {
+    // Inputs contain optional bias
+    const auto& bias_name = input_defs[2]->Name();
+    if (!Contains(initializers, bias_name)) {
+      LOGS(logger, VERBOSE) << "The bias must be a constant initializer.";
+      return false;
+    }
+  }
+
+  const auto& output_defs = node.OutputDefs();
+  if (output_defs.size() != 1) {
```

Comment:
This condition is referring to https://github.com/Honry/onnxruntime/blob/stable-diffusion/onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorLayerNormalization.cpp#L258
```c++
void CALLBACK QueryLayerNormalization(IMLOperatorSupportQueryContextPrivate* context, /*out*/ bool* isSupported)
{
    *isSupported = context->GetOutputCount() == 1;
}

```

Reply:
Yeah, that's fine. I've never actually seen those two outputs used in a model anyway for prediction/inference (probably only applicable to training).